REST/internal/auth/service.go

```diff
@@ -0,0 +1,87 @@
+package auth
+
+import (
+	"REST/internal/models"
+	"errors"
+
+	"golang.org/x/crypto/bcrypt"
+	"gorm.io/gorm"
+)
+
+type AuthService struct {
+	db *gorm.DB
+}
+
+// New creates a new authentication service instance.
+func New(db *gorm.DB) *AuthService {
+	return &AuthService{db: db}
+}
+
+// Register a new user.
+func (s *AuthService) Register(email, password, userType string) (*models.Base, error) {
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return nil, err
+	}
+
+	var base models.Base
+	switch userType {
+	case "student":
+		student := models.Student{
+			Email:    email,
+			Password: string(hashedPassword),
+		}
+		if err := s.db.Create(&student).Error; err != nil {
+			return nil, err
+		}
+		base = student.Base
+	case "professor":
+		professor := models.Professor{
+			Email:    email,
+			Password: string(hashedPassword),
+		}
+		if err := s.db.Create(&professor).Error; err != nil {
+			return nil, err
+		}
+		base = professor.Base
+	default:
+		return nil, errors.New("invalid user type")
+	}
+
+	return &base, nil
+}
+
+// Login checks the credentials and returns the user if they're correct.
+func (s *AuthService) Login(email, password, userType string) (*models.Base, error) {
```

Comment:
For now I have no problem for this but later when we develop front cookie will be useful later (create next task or do it now)